### PR TITLE
Rename adapter and serializer methods to match the new store methods

### PR DESCRIPTION
**Pull request by bmac**
`Adapter#find` -> `Adapter#findRecord` and `Adapter#findQuery` -> `Adapter#query`
`BuildURLMixin#urlForFind` -> `BuildURLMixin#urlForFindRecord` and `BuildURLMixin#urlForFindQuery` -> `BuildURLMixin#urlForQuery`
`Serializer#extractFind` -> `Serializer#extractFindRecord` and `Serializer#findQuery` -> `Serializer#extractQuery`

### CODE REVIEW

**Comment by bmac** [comment]
Ping @amk221. I told you I didn't think this change was happening, I was wrong.

**Comment by amk221** [comment]
:) thanks for the heads up. I predicted the future

File: packages/ember-data/lib/serializers/json-serializer.js [hunk 1]
```diff
@@ -1249,8 +1249,31 @@ export default Serializer.extend({
...
+    @param {Object} payload
+    @param {(String|Number)} id
+    @param {String} requestType
+    @return {Array} array An array of deserialized objects
+  */
+  extractQuery: function(store, typeClass, payload, id, requestType) {
+    var jsonExtractFindQuery = JSONSerializer.prototype.extractFindQuery;
+    if (jsonExtractFindQuery !== this.extractFindQuery) {
+      Ember.deprecate('Serializer#extractFindQuery has been deprecated and renamed to `extractQuery`.');
+      return this.extractFindQuery(store, typeClass, payload, id, requestType);
```
Comment (bmac) [inline packages/ember-data/lib/serializers/json-serializer.js]: `return this.extractFindQuery(store, typeClass, payload, id, 'findQuery'); `

File: packages/ember-data/lib/serializers/json-serializer.js [hunk 2]
```diff
@@ -1347,10 +1370,35 @@ export default Serializer.extend({
...
+    @param {Object} payload
+    @param {(String|Number)} id
+    @param {String} requestType
+    @return {Object} json The deserialized payload
+  */
+  extractFindRecord: function(store, typeClass, payload, id, requestType) {
+    var jsonExtractFind = JSONSerializer.prototype.extractFind;
+    if (jsonExtractFind !== this.extractFind) {
+      Ember.deprecate('Serializer#extractFind has been deprecated and renamed to `extractFindRecord`.');
+      return this.extractFind(store, typeClass, payload, id, requestType);
```
Comment (bmac) [inline packages/ember-data/lib/serializers/json-serializer.js]: `return this.extractFind(store, typeClass, payload, id, 'find');`

File: packages/ember-data/lib/adapters/build-url-mixin.js [hunk 3]
```diff
@@ -46,17 +46,17 @@ export default Ember.Mixin.create({
...
     @return {String} url
   */
   buildURL: function(modelName, id, snapshot, requestType, query) {
     switch (requestType) {
-      case 'find':
-        return this.urlForFind(id, modelName, snapshot);
+      case 'findRecord':
+        return this.urlForFindRecord(id, modelName, snapshot);
       case 'findAll':
         return this.urlForFindAll(modelName);
```
Comment (igorT) [inline packages/ember-data/lib/adapters/build-url-mixin.js]: This breaks backwards compatibility as well right? We should keep the old hooks in for 1.13 no?

File: packages/ember-data/lib/serializers/json-serializer.js [hunk 4]
```diff
@@ -1249,8 +1249,31 @@ export default Serializer.extend({
...
+
+    @method extractQuery
+    @param {DS.Store} store
+    @param {DS.Model} typeClass
+    @param {Object} payload
+    @param {(String|Number)} id
+    @param {String} requestType
+    @return {Array} array An array of deserialized objects
+  */
+  extractQuery: function(store, typeClass, payload, id, requestType) {
```
Comment (igorT) [inline packages/ember-data/lib/serializers/json-serializer.js]: This shouldn't be a thing. This is the old api.

Comment (igorT) [inline packages/ember-data/lib/serializers/json-serializer.js]: I guess if someone is migrating might be nice though.

**Comment by igorT** [comment]
I couldn't find any new tests for the deprecated behavior, are they there or do we still need to add them?

**Comment by igorT** [comment]
This fails on travis when optional features are turned on

File: packages/ember-data/lib/adapters/build-url-mixin.js (hunk 3 above)
Comment (bmac) [inline packages/ember-data/lib/adapters/build-url-mixin.js]: `urlForFindRecord` calls `urlForFind` if it looks like it has been overwritten  https://github.com/emberjs/data/pull/3306/files#diff-fffd24b16fb23892703ed3ab601e6c80R123

**Pull request merged**